### PR TITLE
Added monitor subnet to SapMonitor model

### PR DESCRIPTION
**Pull request by PakDLiu**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by ArcturusZhang** [comment]
Hi @PakDLiu could you please provide some information about the `breaking change`? Adding properties to response is identified as a breaking change by the breaking change tool. 
The new field subnet is added only to responses, does this mean that field should be readonly?

**Comment by PakDLiu** [comment]
Yes, I added the new field: monitorSubnet.
It is required in the request.
It should also be returned in the response

File: specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2020-02-07-preview/hanaonazure.json [hunk 1]
```diff
@@ -617,6 +617,10 @@
           "readOnly": true,
           "type": "string",
           "description": "The version of the payload running in the Collector VM"
+        },
+        "monitorSubnet": {
```
Comment (majastrz) [inline specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2020-02-07-preview/hanaonazure.json:621]: >monitorSubnet [](start = 9, length = 13)

Is this property required?

File: specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2020-02-07-preview/examples/SapMonitors_Create.json [hunk 2]
```diff
@@ -16,7 +16,8 @@
         "enableCustomerAnalytics": true,
         "logAnalyticsWorkspaceArmId": "",
         "logAnalyticsWorkspaceId": "",
-        "logAnalyticsWorkspaceSharedKey": ""
+        "logAnalyticsWorkspaceSharedKey": "",
```
Comment (majastrz) [inline specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2020-02-07-preview/examples/SapMonitors_Create.json:19]: >"" [](start = 42, length = 2)

Can you provide example resource IDs for all the properties that are being specified. `""` is not really helpful to users. (This comment applies to all examples and properties in the examples.)

File: specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2020-02-07-preview/hanaonazure.json (hunk 1 above)
Comment (PakDLiu) [reply specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2020-02-07-preview/hanaonazure.json:621]: yes

Comment (majastrz) [reply specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2020-02-07-preview/hanaonazure.json:621]: Then this is a breaking change that should be done in a new API version.

---
In reply to: [402059679](https://github.com/Azure/azure-rest-api-specs/pull/8915#discussion_r402059679) [](ancestors = 402059679)

Comment (PakDLiu) [reply specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2020-02-07-preview/hanaonazure.json:621]: This is still a preview API version and the API is still in flux.
We also know that no one outside of our team is using this API yet.

Comment (majastrz) [reply specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2020-02-07-preview/hanaonazure.json:621]: Got it. That's fine then. 

In general, I would recommend working in a development branch in the public or private Swagger repo when the API is still in flux. What's in `master` branch in this repo is considered public and available for use regardless of the `-preview` suffix on the API version.

---
In reply to: [402062621](https://github.com/Azure/azure-rest-api-specs/pull/8915#discussion_r402062621) [](ancestors = 402062621)

**Review by majastrz — APPROVED** [review]
Please update examples to make them more meaningful, but the change looks good to me. Signed off from ARM side.

File: specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2020-02-07-preview/examples/SapMonitors_Create.json (hunk 2 above)
Comment (PakDLiu) [reply specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2020-02-07-preview/examples/SapMonitors_Create.json]: done

File: specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2020-02-07-preview/hanaonazure.json (hunk 1 above)
Comment (PakDLiu) [reply specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2020-02-07-preview/hanaonazure.json:621]: I see, thanks!
Will do that in the future

**Comment by ArcturusZhang** [comment]
Hi @PakDLiu could you please resolve the CI failures? (Except the breaking change since it is acceptable for a preview API) then this will be good to merge, thanks

**Comment by PakDLiu** [comment]
Didn't notice there were other errors.
Fixed

**Review by ArcturusZhang — APPROVED** [review]
LGTM 👍

**Pull request merged**